### PR TITLE
Bumped mypy to 1.11.0 and jaxlib to 0.4.31 on the CI

### PR DESCRIPTION
**Pull request by superbobry**

### CODE REVIEW

File: jax/_src/scipy/signal.py [hunk 1]
```diff
@@ -753,19 +753,20 @@ def pad(x, n, axis=-1):
       y_arr = jnp.concatenate((y_arr, jnp.zeros_like(x, shape=(*y_arr.shape[:-1], nadd))), axis=-1)
 
   # Handle detrending and window functions
-  if not detrend_type:
+  if isinstance(detrend_type, bool):
```
Comment (superbobry) [inline jax/_src/scipy/signal.py:756]: This is not identical, but I think that was the intent?

Comment (jakevdp) [reply jax/_src/scipy/signal.py:756]: I don't think this is a safe change; for example `np.False_` is not an instance of `bool`.

Also, this is the incorrect output for `detrend=True`, which is not a valid argument. In the current implementation, `True` would be passed to `detrend`, which checks the validity and raises an error. In the new implementation, `True` would be treated the same as `False`.

Comment (superbobry) [reply jax/_src/scipy/signal.py]: Changed as discussed offline. PTAL.

File: jax/_src/scipy/signal.py [hunk 2]
```diff
@@ -753,19 +753,20 @@ def pad(x, n, axis=-1):
...
+  elif callable(detrend_type):
+    if axis != -1:
+      # Wrap this function so that it receives a shape that it could
+      # reasonably expect to receive.
+      def detrend_func(d):
+        d = jnp.moveaxis(d, axis, -1)
+        d = detrend_type(d)
+        return jnp.moveaxis(d, -1, axis)
+    else:
+      detrend_func = detrend_type
```
[the remark this answers is not in the document]
Comment (superbobry) [reply jax/_src/scipy/signal.py:769]: Ack.

**Review by jakevdp — APPROVED** [review]
Looks great!

**Pull request merged**